src/components/Input/Input.tsx

```diff
@@ -67,12 +67,11 @@ export const Input = React.forwardRef<HTMLInputElement, Props>(
     ) => {
         const fallbackId = React.useId();
         const idRef = id || fallbackId;
-        const [hasText, setHasText] = React.useState<boolean>(!!value);
+        const { onChange, style } = rest;
 
         const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-            setHasText(!!e.target.value);
-            if (rest.onChange) {
-                rest.onChange(e);
+            if (onChange) {
+                onChange(e);
```

Comment:
Looks like now this function does what the native callback would do, so maybe we can just leave it out, no?